internal/server/storage/minio/data.go

```diff
@@ -0,0 +1,231 @@
+package minio
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+
+	fl "github.com/igortoigildin/goph-keeper/pkg/file"
+	"github.com/igortoigildin/goph-keeper/pkg/logger"
+	"github.com/minio/minio-go/v7"
+	"github.com/minio/minio-go/v7/pkg/credentials"
+	"go.uber.org/zap"
+)
+
+const (
+	accessKey = "minioaccesskey"
+	secretKey = "miniosecretkey"
+	useSSL    = false
+	endpoint  = "localhost:9000"
+)
+
+type DataRepository struct{}
+
+func NewRepository() *DataRepository {
+	return &DataRepository{}
+}
+
+func (d *DataRepository) SaveFile(ctx context.Context, file *fl.File, login string, id string, meta string) error {
+	client, err := minio.New(endpoint, &minio.Options{
+		Creds:  credentials.NewStaticV4(accessKey, secretKey, ""),
+		Secure: useSSL,
+	})
+	if err != nil {
+		logger.Error("error while creating minio client: ", zap.Error(err))
+
+		return fmt.Errorf("error instantiating Minio client with options: %w", err)
+	}
+
+	// Define the file to upload and the destination bucket
+	objectName := id    // The name for the object in MinIO
+	bucketName := login // Bucket name in MinIO
+
+	// Ensure the bucket exists (or create it)
+	err = client.MakeBucket(ctx, bucketName, minio.MakeBucketOptions{})
+	if err != nil {
+		if exists, errBucketExists := client.BucketExists(context.Background(), bucketName); errBucketExists == nil && exists {
```

Comment:
```suggestion
		if exists, errBucketExists := client.BucketExists(ctx, bucketName); errBucketExists == nil && exists {
```
не очень понятно, почему именно здесь контекст берется не от запроса, а из воздуха.

Reply:
Обновил